mod_modular_vanilla/hooks/skills/actives/footwork.nut

```diff
@@ -0,0 +1,18 @@
+::ModularVanilla.MH.hook("scripts/skills/actives/footwork", function(q) {
+	// Convert the vanilla method of "setting" certain fields to instead be incremental changes
+	q.onAfterUpdate = @() function( _properties )
+	{
+		if (_properties.IsFleetfooted)
+		{
+			this.m.FatigueCostMult *= 0.5;
+
+			if (this.getContainer().hasSkill("effects.goblin_grunt_potion"))
+			{
+				if (this.m.ActionPointCost > 2)
+				{
+					this.m.ActionPointCost -= 2;
```

Comment:
This way you can discount Footwork down to 1 AP if it started at 3 AP cost. But if Footwork was  2 AP cost then this potion has no effect.
Instead I suggest using a `this.m.ActionPointCost = ::Math.max(1, this.m.ActionPointCost - 2)` instead.

Reply:
This mod is meant to emulate vanilla behavior as close as possible and in vanilla the goblin_grunt_potion tooltip says that it reduces the AP cost of Footwork to 2. This mod shouldn't change that design and instead that should be the realm of other mods. What do you think?

Reply:
I think your suggestion makes sense so I'll go with that.